mxfusion/components/variables/variable.py

```diff
@@ -40,10 +40,12 @@ class Variable(ModelComponent):
     """
     def __init__(self, value=None, shape=None, transformation=None, isInherited=False, initial_value=None):
         super(Variable, self).__init__()
-
-        # TODO If no shape we assume a scalar but this could be incorrect if we really just mean the shape is unknown.
-        self.shape = shape if shape is not None else (1,)
-        self.attributes = [s for s in self.shape if isinstance(s, Variable)]
+        self.shape = shape  # When initializing array params, shape is updated if passed as None
```

Comment:
```suggestion
        self.shape = shape  # Shape is inferred at inference runtime if passed as None
```

Reply:
Shape update happens when Variable is created. Doesn't inference in a different part of the code?